src/main/java/de/dennisguse/opentracks/util/PreferencesUtils.java

```diff
@@ -39,12 +48,34 @@
 
     private final static String TAG = PreferencesUtils.class.getSimpleName();
 
+    private static final int PREFERENCES_VERSION = 1;
+
     private PreferencesUtils() {
     }
 
     @Deprecated //Should only be used to get a sharedPreference for more than one interaction!
     public static SharedPreferences getSharedPreferences(Context context) {
-        return PreferenceManager.getDefaultSharedPreferences(context);
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        int lastVersion = getInt(sharedPreferences, context, R.string.prefs_last_version_key, 0);
+        if (PREFERENCES_VERSION > lastVersion) {
+            onUpgrade(sharedPreferences, context, PREFERENCES_VERSION);
+            setInt(sharedPreferences, context, R.string.prefs_last_version_key, PREFERENCES_VERSION);
+        }
+        return sharedPreferences;
+    }
+
+    private static void onUpgrade(SharedPreferences sharedPreferences, Context context, int toVersion) {
```

Comment:
can we extract this into a separate file?
e.g., PreferenceOpenHelper (similar to CustomSQLiteOpenHelper)?